fs/mmap/fs_rammap.c

```diff
@@ -235,7 +237,6 @@ int rammap(FAR struct file *filep, FAR struct mm_map_entry_s *entry,
 
   /* Add the buffer to the list of regions */
 
-  entry->vaddr = rdbuffer;
   entry->priv.i = kernel;
   entry->munmap = unmap_rammap;
 
```

Comment:
the next ret check with goto errout_with_region is actually wrong as well; the pointer to rdbuffer is no longer pointing to the allocated area, but is still freed there

Reply:
In the "free" below we should use entry->vaddr, since that stores the original allocated ptr